cis/processor.py

```diff
@@ -0,0 +1,127 @@
+"""Iterative Stream Processor plugin for handling steps during data storage."""
+import json
+import logging
+import os
+
+from cis import user
+
+from cis.libs import encryption
+from cis.libs import streams
+from cis.libs import validation
+
+logger = logging.getLogger(__name__)
+
+class OperationDelegate(object):
+    def __init__(self, boto_session=None, publisher=None, signature=None, encrypted_profile_data=None):
+        self.boto_session = boto_session
+        self.dry_run = True
+        self.decryptor = encryption.Operation(boto_session=boto_session)
+        self.encrypted_profile_data = encrypted_profile_data
+        self.kinesis_client = None
+        self.publisher = publisher
+        self.signature = signature
+        self.stage = self._get_stage()
+        self.user = None
+
+    def run(self):
+        # Determine what stage of processing we are in and call the corresponding functions.
+
+        self.decrytped_profile = json.loads(self._decrypt_profile_packet())
+
+        if self.stage == 'validator':
+            logger.info('Validator logic activated.')
+            res = self._validator_stage()
+
+            if res is True:
+                logger.info('Payload valid sending to kinesis.')
+                res = streams.Operation(
+                    boto_session=self.boto_session,
+                    publisher=self.publisher,
+                    signature=self.signature,
+                    encrypted_profile_data=self.encrypted_profile_data
+                ).to_kinesis()
+
+                if res['ResponseMetadata']['HTTPStatusCode'] == 200:
+                    res = True
+        elif self.stage == 'streamtoidv':
+            res = self._vault_stage()
+        elif self.stage == 'idvtoauth0':
+            # TBD fold in the authzero logic from idvtoauth0 in cis_functions
```

Comment:
nit: s/TBD/FIXME. i usually just go with smth like 'XXX FIXME TBD INSERTURSHERE' so that everyone's favorite IDE picks these up